x-pack/plugin/security/src/internalClusterTest/java/org/elasticsearch/xpack/security/ReloadRemoteClusterCredentialsIT.java

```diff
@@ -0,0 +1,314 @@
+    private static final String CLUSTER_ALIAS = "my_remote_cluster";
+
+    @BeforeClass
+    public static void disableInFips() {
+        assumeFalse(
+            "Cannot run in FIPS mode since the keystore will be password protected and sending a password in the reload"
+                + "settings api call, require TLS to be configured for the transport layer",
+            inFipsJvm()
+        );
+    }
+
+    @Override
+    public String configRoles() {
+        return org.elasticsearch.core.Strings.format("""
+            user:
+              cluster: [ "ALL" ]
+              indices:
+                - names: '*'
+                  privileges: [ "ALL" ]
+              remote_indices:
+                - names: '*'
+                  privileges: [ "ALL" ]
+                  clusters: ["*"]
+            """);
+    }
+
+    @Override
+    public void tearDown() throws Exception {
+        try {
+            clearRemoteCluster();
+            super.tearDown();
+        } finally {
+            ThreadPool.terminate(threadPool, 10, TimeUnit.SECONDS);
+        }
+    }
+
+    private final ThreadPool threadPool = new TestThreadPool(getClass().getName());
+
+    public void testReloadRemoteClusterCredentials() throws Exception {
+        final String credentials = randomAlphaOfLength(42);
+        writeCredentialsToKeyStore(credentials);
+        final RemoteClusterCredentialsManager clusterCredentialsManager = getInstanceFromNode(TransportService.class)
+            .getRemoteClusterService()
+            .getRemoteClusterCredentialsManager();
+        // Until we reload, credentials written to keystore are not loaded into the credentials manager
+        assertThat(clusterCredentialsManager.hasCredentials(CLUSTER_ALIAS), is(false));
+        reloadSecureSettings();
+        assertThat(clusterCredentialsManager.resolveCredentials(CLUSTER_ALIAS), equalTo(credentials));
+
+        // Check that credentials get used for a remote connection, once we configure it
+        final BlockingQueue<Map<String, String>> capturedHeaders = ConcurrentCollections.newBlockingQueue();
+        try (MockTransportService remoteTransport = startTransport("remoteNodeA", threadPool, capturedHeaders)) {
+            final TransportAddress remoteAddress = remoteTransport.getOriginalTransport()
+                .profileBoundAddresses()
+                .get("_remote_cluster")
+                .publishAddress();
+
+            configureRemoteCluster(remoteAddress);
+
+            // Run search to trigger header capturing on the receiving side
+            client().search(new SearchRequest(CLUSTER_ALIAS + ":index-a")).get();
+
+            assertHeadersContainCredentialsThenClear(credentials, capturedHeaders);
+
+            // Update credentials and ensure they are used
+            final String updatedCredentials = randomAlphaOfLength(41);
+            writeCredentialsToKeyStore(updatedCredentials);
+            reloadSecureSettings();
```

Comment:
I'm not sure I fully understand the relationship between connection profile and the need to rebuild the connection. Once the connection is established (TCP + handshake) it should remain open and if need to re-establish will use the new credential. Per request will use the new credential. I assumed that the rebuilding the connection was for 1) ensure we disconnect any old connections that we established with the old credential (more clean up than anything) 2) so that we can error or at least warn if the new credential is not valid on the reload call.  I can't remember if it is possible to explictly change the underlying transport profile (https://www.elastic.co/guide/en/elasticsearch/reference/current/modules-network.html#transport-profiles) but we intentionally didn't document the one used for RCS 2.0 to discourage direct configuration there and any related options that influence the underlying transport profile are exposed as dedicated settings and are not related to the credential reload.